doc/src/challenges/0004-btree-node.md

```diff
@@ -0,0 +1,69 @@
+# Challenge 4: Verify the memory safety of the `alloc::collections::btree::node` module
+
+- **Status:** Open
+- **Tracking Issue:** [Link to issue](https://github.com/model-checking/verify-rust-std/issues/25)
+- **Start date:** *2024-07-01*
+- **End date:** *2024-12-10*
+
+-------------------
+
+
+## Goal
+
+Verify the memory safety of the [`alloc::collections::btree::node` module](https://github.com/rust-lang/rust/blob/c290e9de32e8ba6a673ef125fde40eadd395d170/library/alloc/src/collections/btree/node.rs).
+This is one of the main modules used for implementing the `BTreeMap` collection, and it includes a lot of unsafe code.
+
+After annotating functions in this module with their safety contract and verifying those contracts, verify that their usage from `BTreeMap` satisfies the contracts.
+
+### Success Criteria
+
+All public functions (especially safe ones) containing unsafe code must be annotated with safety contracts and the contracts have been verified, e.g.:
```

Comment:
Safe functions don't need any `safety contract` though. They have an implicit contract that `self` must be safe which we can explicitly add today, but that's about it.

Reply:
Updated the wording.